server/server.js

```diff
@@ -40,13 +35,36 @@ app.post("/setPiece", async (req, res) => {
   }
 
   currentColumns[x] = y - 1; // so row moves up by 1 row
-  console.log("BOARD", boardArr);
   res.status(200);
   res.send();
 });
 
 app.get("/fetchGameStateVariables", (req, res) => {
-  res.json({ boardArr, currentPlayer, currentColumns }); // sending boardArr, currentPlayer and currentColumns variables to client
+  res.json({
+    boardArr,
+    currentPlayer,
+    currentColumns,
+  }); // sending boardArr, currentPlayer and currentColumns variables to client
+});
+
+app.post("/newGame", async (req, res) => {
+  const newBoard = [
+    [null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null],
+  ];
+
+  const resetCurrentColumns = [5, 5, 5, 5, 5, 5, 5];
+
+  currentColumns = [...resetCurrentColumns];
+
+  boardArr = [...newBoard];
+
```

Comment:
Do you need to reset who the current player is?

Reply:
No, when a player wins, the current player is already changed due to the setPiece function on the client and server.

E.g. If it's turn red, and red makes the winning move, the cursor will turn yellow and the board game will say "RED WINS!". So when the new game button is clicked, the cursor is already yellow.